Rename PetSet to StatefulSet

soltysh opened the pull request:
@ncdc ptal

soltysh commented on pkg/cmd/server/bootstrappolicy/infra_sa_policy.go:
```diff
@@ -65,8 +65,8 @@ const (
 	// TODO can we just rename these or how do we handle upgrades?
-	InfraPetSetControllerServiceAccountName = "pet-set-controller"
-	PetSetControllerRoleName                = "system:pet-set-controller"
+	InfraStatefulSetControllerServiceAccountName = "pet-set-controller"
```
@deads2k @liggitt can we rename the SA here? What's the recommended approach for those? Currently (as of 1.5) there's no such resource as PetSet, no migration was provided, either. Any ideas, suggestions how to proceed?

deads2k replied:
> @deads2k @liggitt can we rename the SA here? What's the recommended approach for those? Currently (as of 1.5) there's no such resource as PetSet, no migration was provided, either. Any ideas, suggestions how to proceed?

Remove the old from here and then specify the new.  The old permissions will rot, but not be removed on customer systems until they run a manual role reconcile, but that's ok.

Also, match names with upstream if possible.

soltysh commented:
[test]

ncdc commented:
Rename `examples/pets` dir to statefulsets?

soltysh commented:
Updated all as suggested by @ncdc and @deads2k. I've also went ahead and removed any petset from bootstrap policy, since there's no way you can get even the old PetSet resources.

soltysh commented:
That previous failure is reported in https://github.com/openshift/origin/issues/12500, but since we previously didn't run StatefulSet in conformance (due to the rename) this wasn't breaking us.

soltysh commented:
Rebased to fix that last conflict.

ncdc commented:
```
FAILURE after 0.527s: test/cmd/basicresources.sh:172: executing 'oc create -f examples/pets/zookeeper/zookeeper.yaml' expecting success: the command returned the wrong error code
```

soltysh commented:
Yeah, I've noticed that one. I'm currently working on fixing #12500 to be able to pass the tests. Without it this can't merge.

soltysh commented:
The 2nd commit should fix #12500, it's setting tests to work properly with hostPath mounts which are used in StatefulSets tests. @smarterclayton fyi

smarterclayton commented:
Yay [merge] thanks

smarterclayton commented:
Did we exclude statefulsets from conformance?  If so, we should put one stateful set test into conformance.

soltysh commented:
> Did we exclude statefulsets from conformance? If so, we should put one stateful set test into conformance.

No, they're part of conformance, at least now since the rename I just did, see [here](https://github.com/openshift/origin/pull/12642/files#diff-21c58c923d55767e64fef3ecfd640ac0R362).

smarterclayton commented:
Great, thanks
